docs/proposals/scheduling/20221227-cpu-static-reservation.md

```diff
@@ -0,0 +1,176 @@
+so for k8s, the 0-11 core is reserved and it cannot be used.
+
+#### Story 2
+As a k8s cluster administrator, set the total amount of resources to be reserved directly via `node.annotation`, just like: `1C2Gi`.
+
+#### Story 3
+As a k8s cluster administrator, set the total number of cpu cores to be reserved via kubelet, and then set the specific cpu cores to be reserved via `node.annotation`, eg: kubelet sets "4C", node.annaion sets the cores with id 0-3 not to be used.
+
+### Implementation Details
+By adding annotation to the node, we specify the resources to be reserved by the koordinator, such as memory, cpu, etc. For cpu, we can specify the total amount of cpu to be reserved, or we can specify which cores to explicitly reserve. 
+```golang
+// resources that koordinator reserved, and you can reserve other resource if needed.
+type KoordReserved struct {
+	// total cpus need to be reserved, just like "2C"
+	CPUNums      string `json:"CPUNums"`
+	// specific cpus need to be reserved, just like "0-6", it means that 0-6 core cannot be used by k8s.
+	specificCPUs string `json:"specificCPUs"`
+	// eg: "1Gi"
+	memory       string `json:"memory"`
+}
+```
+#### koordlet   
+1. When koordlet reports `NodeResourceTopology`, it updates the explicitly reserved cores in node.annntation to `NodeResourceTopology.annotation["node.koordinator.sh/reserved-cpus"]`, so that the `runtimehook` can ignore these cores when fetching the shared-pool from nodetopo.
+
+2. cpu supress:  
+it will evict some pod whose qos is not lse. we should ignore the cpus reserved by node when supress with cpuset policy.  
+
+```
+cpuset(lsr) = processor(lsr) - cpusReservedByNode
+cpuset(!lse/lser) = processor(!lse/lser) - cpusReservedByNode
+``` 
+
+3. runtimehooks:  
+it will ignored the cpus reserved by `node.annotations` when create container. 
+```
+cpus(allocatable) = cpus(shared-pools) - cpus(reservedByNode)
+```  
+  
+<font color=Chocolate>*attention:*</font> The cpu occupied by the lse pod on the node is not considered here for now, but it will be removed later in the cpuset suppression phase.
+
+
+#### koord-manager  
+should update batch-resource on node, just like `batch-memory`,`batch-cpu`.the implementation of this part depends on the `nodeResource`  controller in koord-manager, so for batch-resource, the `batchResourceFit` plugin can take into account the resources reserved by the node.
+
+```
+Node(BE).Alloc = Node.Total - Node.Reserved - node.anno.reserved - System.Used - Pod(LS).Used
+```
+
+#### koord-scheduler
+
+- for memory:  
+  node.alloc - node.requested.memory - node.annotation[koord-reserved].memory >= pod.request
+- for cpu:  
+  1. both nums and specific are declared on node.anno:   
+     nums will be ignored.
+  2. only nums:  
+     node.alloc.cpu - node.requested.cpu - node.annotation[koord-reserved].cpu >= pod.request.cpu
+  3. only specific cpus:  
+     pod.anno[specific-cpus] $\bigcap$ node.anno[koord-specific-cpus-reserved] = $\phi$   
```

Comment:
Don't understand what specific-cpus is. Please add more details.

Reply:
just like this demo: https://github.com/koordinator-sh/koordinator/blob/main/pkg/koordlet/statesinformer/states_noderesourcetopology_test.go#L196-L210

Reply:
OK, I got it. Skip the reserved CPUs during koord-scheduler allocating CPUs.